examples/src/main/java/kafka/examples/Consumer.java

```diff
@@ -63,33 +63,36 @@ public Consumer(final String topic,
         this.numMessageToConsume = numMessageToConsume;
         this.messageRemaining = numMessageToConsume;
         this.latch = latch;
+        this.isRunning = true;
     }
 
     KafkaConsumer<Integer, String> get() {
         return consumer;
     }
 
     @Override
+    public void run() {
+        try {
+            do {
+                doWork();
+            } while (isRunning && messageRemaining > 0);
```

Comment:
Why should we need `isRunning`? From the usage, we will wait for 5 mins for the `latch` completed, if not, `TimeoutException` will be thrown, so no other thread will call `shutdown` during it's running. Is my understanding correct?